### ISSUE TITLE
aws-appsync errors calling FetchResult

### ISSUE THREAD

**Issue by dexster** [awslabs/aws-mobile-appsync-sdk-js#553]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
aws-appsync@3.0.3 has these errors;

> ERROR in node_modules/aws-appsync-auth-link/lib/auth-link.d.ts:24:55 - error TS2707: Generic type 'FetchResult' requires between 0 and 2 type arguments.
> 
> 24     request(operation: any, forward: any): Observable<import("apollo-link").FetchResult<{
>                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 25         [key: string]: any;
> 26     }, Record<string, any>, Record<string, any>>>;

> node_modules/aws-appsync/lib/client.d.ts:63:98 - error TS2707: Generic type 'FetchResult' requires between 0 and 2 type arguments.
> 
> 63     mutate<T, TVariables = OperationVariables>(options: MutationOptions<T, TVariables>): Promise<import("apollo-link").FetchResult<T, Record<string, any>, Record<string, any>>>;
> 

**Which versions and which environment (browser, react-native, nodejs) / OS are affected by this issue? Did this work in previous versions?**

This is in an angular 9 CLI project

Errors occur with:

> aws-appsync 3.0.3
> aws-appsync-auth-link 2.0.2

I reverted to:

> aws-appsync 3.0.2
> aws-appsync-auth-link 2.0.1

and errors dissappear

**Comment by RuggeroVisintin** [comment]
Same problem.
Solved with the same solution just now

**Comment by sp-eric-pinto** [comment]
The issue is mainly that apollo-link supports `FetchResult<x,y,z>`only since version `apollo-link@^1.2.5` and it's fixed to `"apollo-link": "1.2.3"` which only accepts 2 parameters

**Comment by JBhavsar-Fintech** [comment]
I am having the same issue.Tried downgrading the versions.still it's not working.

**Comment by jts-mjanda** [comment]
The workaround to this found in this issue (just about the same problem) is here:
https://github.com/awslabs/aws-mobile-appsync-sdk-js/issues/559

**Comment by dexster** [comment]
When will this be fixed? It's been almost 3 months since it was reported.

It's simple  to reproduce. Create a new app using angular-cli:

`ng new testAppSync`

`cd testAppSync`

`npm install aws-appsync`

Paste this into your app.component.ts and `ng serve`

```

import {Component, OnInit} from '@angular/core';
import AWSAppSyncClient from 'aws-appsync/lib';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {

  ngOnInit() {
const appsyncClient = new AWSAppSyncClient({});
  }
}
```

**Comment by jts-mjanda** [comment]
I personally have given up on this package. I went with the aws-amplify package and a configured apollo client. If you're waiting for this to be resolved, I would encourage others to do the same.

**Comment by dexster** [comment]
I've been using workarounds for the past few months. I need offline sync and complex object support which amplify does not provide out the box and the AWS documentation points to this library, so it needs to be resolved

**Comment by pickfire** [comment]
This is too late, the fix was deployed after a release version was published. Shouldn't these kind of stuff be fixed in alpha or beta release? Why even publish a broken release?

**Comment by dexster** [comment]
I'm still using the old versions until they update their package

**Comment by dexster** [comment]
It seems to finally be fixed in 4.0.1, 5 months later. This is incredibly poor service from AWS
